text/src/autogluon/text/automm/configs/environment/default.yaml

```diff
@@ -3,7 +3,7 @@ env:
   num_nodes: 1
   batch_size: 128  # this is a desired batch size; pl trainer will accumulate gradients when per step batch is smaller.
   per_gpu_batch_size: 8  # training per gpu batch size
-  per_gpu_batch_size_evaluation: 64  # evaluation per gpu batch size
```

Comment:
You should keep the old key. Lots of users are using the key. You may consider to add a new key and change the default value of the old key to null.

Reply:
We already make this change backward compatible here https://github.com/awslabs/autogluon/pull/1721/files#diff-9a3106c0c6088dce9389c316f54a34d170a1080af4121fba0c194359f676d89eR935

It's kind of misleading if we keep the old key but not use it.

Reply:
What's our general design guideline for adding new keys?

Reply:
Also, it will be problematic if the users are revising the `per_gpu_batch_size_evaluation` via overrides. They will be adding a new key rather than updating an existing key.

Reply:
Adding a new key is designed to raise an exception in https://github.com/awslabs/autogluon/blob/c03d7ba762b1982113b797fd9e8ee3559d2c3a43/text/src/autogluon/text/automm/utils.py#L217

Reply:
Since this PR will go into 0.4.1, we need to make it as backward compatible as possible.

Reply:
Let's say we still keep `per_gpu_batch_size_evaluation ` but not actually use it. If users override `per_gpu_batch_size_evaluation `, it doesn't behave as they expected. Is this OK?

Reply:
We may just change the default value to null.

Reply:
Done.